api/src/test/java/marquez/OpenLineageIntegrationTest.java

```diff
@@ -42,6 +48,35 @@
         EVENT_NAMESPACE_NAMING);
   }
 
+  @Test
+  public void testSendOpenLineageBadArgument() throws IOException {
+    // Namespaces can't have semi-colons, so this will get rejected
+    String badNamespace =
+        "sqlserver://myhost:3342;user=auser;password=XXXXXXXXXX;database=TheDatabase";
+    LineageEvent event =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job("namespace", "job_name", null),
+            List.of(new LineageEvent.Dataset(badNamespace, "the_table", null)),
+            Collections.emptyList(),
+            "the_producer");
+
+    final CompletableFuture<Integer> resp =
+        this.sendLineage(Utils.toJson(event))
+            .thenApply(HttpResponse::statusCode)
+            .whenComplete(
+                (val, error) -> {
+                  if (error != null) {
+                    Assertions.fail("Could not complete request");
+                  }
+                });
```

Comment:
Aren't we expecting an error here? `Assertions.fail()` on error doesn't make sense to me but test seems to be passing. Am I misunderstanding something?

Reply:
Seems to be an error only when there is no response from the server- e.g., can't connect

Reply:
Oh I see.